File: tests/xcalablemp/not-implemented-yet/global-view/template/F/template_fix4.f90
      program main
      include 'xmp_lib.h'
      integer,parameter:: N=1000
!$xmp nodes p(*)
!$xmp template t(:)
!$xmp distribute t(gblock(*)) onto p
      integer s, procs, remain
      integer,allocatable:: a(:)
      integer,allocatable:: m(:)
!$xmp align a(i) with t(i)
      character(len=2) result

      procs = xmp_num_nodes()
      allocate(m(procs))
      
      remain = N
      do i=1, procs-1
         m(i) = remain/2
         remain = remain-m(i)
      enddo
      m(procs) = remain

!$xmp template_fix (gblock(m)) t(N)
      allocate(a(N))

!$xmp loop (i) on t(i)
      do i=1, N
         a(i) = i
      enddo
         
      s = 0
!$xmp loop (i) on t(i) reduction(+: s)
      do i=1, N
         s = s+a(i)
      enddo

      result = 'OK'
      if(s .ne. 500500) then
         result = 'NG'
      endif

      print *, xmp_node_num(), 'testp004.f ', result
      deallocate(a, m)

      end

File: tests/xcalablemp/not-implemented-yet/global-view/template/C/template_fix4.c
#include<xmp.h>
#include<mpi.h>
#include<stdio.h>
#include<stdlib.h>     
static const int N=1000;
#pragma xmp nodes p(*)
#pragma xmp template t(:)
#pragma xmp distribute t(gblock(*)) onto p
int i,j,s,procs,remain;
int *a,*m;
#pragma xmp align a[i] with t(i)
char *result;
int main(void){
   procs=xmp_num_nodes();
   m=(int *)malloc(sizeof(int) * procs);
   remain=N;
   for(i=1;i<procs;i++){
      m[i]=remain/2;
      remain=remain-m[i];
   }
   m[procs]=remain;
#pragma xmp template_fix(block(m)) t(N)
   a=(int *)malloc(sizeof(int) * N);
#pragma xmp loop (i) on t(i)
   for(i=0;i<N;i++){
      a[i]=i;
   }
   s=0;
#pragma xmp loop (i) on t(i) reduction(+:s)
   for(i=0;i<N;i++){
      s+=a[i];
   }
   result="OK";
   if(s != 499500){
      result = "NG";
   }
   printf("%d %s %s\n",xmp_node_num(),"testp004.c",result);
   free (a);
   free (m); 
}
      
         
      
   
